airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSourceOperations.java

```diff
@@ -164,4 +172,53 @@ public JsonSchemaType getAirbyteType(final JDBCType jdbcType) {
     };
   }
 
+  protected void putTimestampWithTimezone(final ObjectNode node, final String columnName, final ResultSet resultSet, final int index)
+      throws SQLException {
+    final OffsetDateTime timestamptz = getObject(resultSet, index, OffsetDateTime.class);
+    final LocalDate localDate = timestamptz.toLocalDate();
+    node.put(columnName, resolveEra(localDate, timestamptz.format(OFFSETDATETIME_FORMATTER)));
+  }
+
+  protected void setTimestampWithTimezone(final PreparedStatement preparedStatement, final int parameterIndex, final String value)
+      throws SQLException {
+    try {
+      final OffsetDateTime offsetDateTime = OffsetDateTime.parse(value, OFFSETDATETIME_FORMATTER);
+      final Timestamp timestamp = Timestamp.valueOf(offsetDateTime.atZoneSameInstant(ZoneOffset.UTC).toLocalDateTime());
+      // Create DateTimeOffset from Timestamp and Offset
+      final DateTimeOffset datetimeoffset = DateTimeOffset.valueOf(timestamp, offsetDateTime.getOffset().getTotalSeconds() / 60);
```

Comment:
This is super nasty stuff. 
Can you add a comment that points to the fact that OffsetDateTime is a java construct, while DateTimeOffset is a sql server construct, and that the offset there is defined in minutes?